core/realm.rs

```diff
@@ -212,12 +216,36 @@ impl JsRealm {
   }
 
   // TODO(andreubotella): `mod_evaluate`, `load_main_module`, `load_side_module`
+}
 
-  pub(crate) fn check_promise_rejections(
+pub struct JsRealmLocal<'s>(v8::Local<'s, v8::Context>);
+impl<'s> JsRealmLocal<'s> {
```

Comment:
This struct makes it unnecessary to upgrade "weak" handles to "global" handles, instead we're upgrading to "local" handles which are cheaper.